llvm/lib/ToolDrivers/llvm-dlltool/DlltoolDriver.cpp

```diff
@@ -215,8 +215,9 @@ int llvm::dlltoolDriverMain(llvm::ArrayRef<const char *> ArgsArr) {
     }
   }
 
-  if (!Path.empty() && writeImportLibrary(Def->OutputFile, Path, Def->Exports,
-                                          Machine, /*MinGW=*/true))
+  if (!Path.empty() &&
+      writeImportLibrary(Def->OutputFile, Path, Def->Exports, std::nullopt,
```

Comment:
I see that you're passing `std::nullopt` here, while in lld/COFF, you pass a real initialized array (which isn't ever used other than passing into this function). Is this because you're planning on using the array in lld?

Reply:
I plan on using it in both llvm-dlltool and lld. I plan to submit llvm-dlltool as soon as this PR settles, lld part will need general ARM64X support in lld, to it will take more time. There was no strong reason to differentiate them in this PR, I guess it was easy to extract the declaration in lld from my WIP tree, while llvm-dlltool will need a small bit of refactoring.